dali/pipeline/operators/displacement/warp_param_provider.h

```diff
@@ -229,19 +229,29 @@ class WarpParamProvider : public InterpTypeProvider, public BorderTypeProvider<B
 
   virtual void GetExplicitPerSampleSize(std::vector<SpatialShape> &out_sizes) const {
     assert(HasExplicitPerSampleSize());
-    const Tensor<CPUBackend> &tensor = ws_->ArgumentInput(size_arg_name_);
-    auto tv = view<const int>(tensor);
+    const TensorList<CPUBackend> &tensor_list = ws_->ArgumentInput(size_arg_name_);
+    const auto &shape = tensor_list.shape();
+    auto tv = view<const int>(tensor_list);
     const int N = num_samples_;
+
+    DALI_ENFORCE(is_uniform(shape), "Output sizes must be passed as uniform Tensor List.");
     DALI_ENFORCE(
-      tv.shape == kernels::TensorShape<>(N, spatial_ndim) ||
-      tv.shape == kernels::TensorShape<>(N * spatial_ndim),
-      "Output sizes must either be a flat array of size num_samples*dim "
-      "or a 2D tensor of size num_samples x dim");
+        (shape.num_samples() == N && shape[0] == kernels::TensorShape<>(spatial_ndim)) ||
+            (shape.num_samples() == 1 && (shape[0] == kernels::TensorShape<>(N, spatial_ndim) ||
+                                          shape[0] == kernels::TensorShape<>(N * spatial_ndim))),
+        "Output sizes must either be a batch of `dim`-sized tensors, flat array of size "
+        "num_samples*dim or a one 2-dim tensor of shape {num_samples, dim}.");
```

Comment:
Why have you replaced 2D with 2-dim? It's confusing in the presence of `dim` here...
```suggestion
        "num_samples*dim or one 2D tensor of shape {num_samples, dim}.");
```

Reply:
Done